nutils/transformseq.py

```diff
@@ -606,7 +623,11 @@ def index_with_tail(self, trans):
         return flatindex, tail
 
 
-class MaskedTransforms(Transforms):
+def MaskedTransforms(parent: Transforms, indices):
```

Comment:
We're not calling `MaskedTransforms` outside this module, so I'm fine with omitting this public function altogether.

Reply:
Fixed. For uniformity I now also reverted `PlainTransforms` and added `transform.iscanonical` to test if the argument is properly preprocessed. This is perhaps a bit controversial - the unit tests are passing with calling `transform.canonical` in most places but I'm not sure if that will be correct in all scenarios. This in turn made me wonder if we actually use the `Singleton` nature for the sequences? If not we can simply process arguments in the constructor.

Reply:
I'm not sure if we need `Singleton` either, but changing `Singleton` into `Immutable` warrants a separate commit and is a bit outside the purpose of this PR.

Reply:
We'll open another PR for this.